src/graph/graph.c

```diff
@@ -1230,13 +1312,16 @@ RelationID Graph_AddRelationType
 ) {
 	ASSERT(g);
 
-	RG_Matrix m;
+	RelationMatrices r;
 	size_t n = Graph_RequiredMatrixDim(g);
+	size_t edge_cap = g->edges->itemCap;
 
-	RG_Matrix_new(&m, GrB_UINT64, n, n);
-
-	array_append(g->relations, m);
+	Delta_Matrix_new(&r.R, GrB_UINT64, n, n, true);
+	Delta_Matrix_new(&r.ME, GrB_BOOL, edge_cap, edge_cap, false);
```

Comment:
I was under the impression we wanted these matrices to be created lazily, (only when the first multi-edge of type Ri is created).